src/components/Sections/Home/DonationPopup.js

```diff
@@ -0,0 +1,177 @@
+import React, { useContext, useState } from 'react'
+import { Button, Grid, Header, Icon, Image, Modal } from 'semantic-ui-react'
+import { DeviceInfoContext } from '../../../helpers/app-contexts';
+import { LANG_ENGLISH, LANG_HEBREW, LANG_RUSSIAN, LANG_SPANISH } from '../../../helpers/consts';
+import * as shapes from '../../shapes';
+import PropTypes from 'prop-types';
+import { getQuery } from '../../../helpers/url';
+import { assetUrl } from '../../../helpers/Api';
+import banner from '../../../images/DonationBanner.jpg'
+import clsx from 'clsx';
+import { isLanguageRtl } from '../../../helpers/i18n-utils';
+
+function DonationPopup({ t, language, location }) {
+  const shouldOpen = () => {
+    const query    = getQuery(location);
+    if (!!query.showPopup)
+      return true;
+    const d = new Date();
+    const firstWeek = d.getDate();
+    const theDay = d.getDay();
+    // First Sunday and Saturday of the week
+    if (firstWeek <= 7 && (theDay == 0 || theDay == 6)) {
+      const popupCountKey = `showDonationPopup_${d.toISOString().split('T')[0]}`;
+      const popupCount = parseInt(localStorage.getItem(popupCountKey) ?? 0);
+      if (popupCount > 1)
+        return false;
+      localStorage.setItem(popupCountKey, (popupCount + 1).toString());
+      return true;
+    }
+
+    return false;
+  }
+
+  const [open, setOpen] = React.useState(shouldOpen())
+  const { isMobileDevice } = useContext(DeviceInfoContext);
+
+  const isRTL = isLanguageRtl(language);
+
+  const { linkLang, utmTerm } = getDonateLinkDetails(language);
+  const link = `https://www.kab1.com/${linkLang}?utm_source=kabbalah_media&utm_medium=popup&utm_campaign=donations&utm_id=donations&utm_term=${utmTerm}&utm_content=popup_link_donate`;
+
+  const getContent = () => {
+    switch (language) {
+      case LANG_HEBREW:
+        return <div dir="rtl">
```

Comment:
if dir from body is not enough?

Reply:
no. because this is modal and outside the main dir="rtl"